otter/integration/tests/test_launch_stack.py

```diff
@@ -0,0 +1,143 @@
+                'name': 'test_launch_stack',
+                'cooldown': 0,
+                'minEntities': 0,
+                'maxEntities': 10
+            },
+            'scalingPolicies': [],
+        }
+
+        self.group = ScalingGroup(group_config=scaling_group_config,
+                                  treq=self.treq,
+                                  pool=self.pool)
+
+        return self.identity.authenticate_user(
+            self.rcs, resources=get_resource_mapping(), region=region)
+
+    def get_stack_list(self):
+        return (self.treq.get(
+                    '%s/stacks' % self.rcs.endpoints['heat'],
+                    headers=headers(str(self.rcs.token)),
+                    params={
+                        'tags': get_stack_tag_for_group(self.group.group_id)},
+                    pool=self.pool)
+                .addCallback(check_success, [200])
+                .addCallback(self.treq.json_content))
+
+    def wait_for_stack_list(self, expected_states, timeout=60, period=10):
+        def check(content):
+            states = pbag([s['stack_status'] for s in content['stacks']])
+            if not (states == expected_states):
+                msg("Waiting for group {} to reach desired group state.\n"
+                    "{} (actual) {} (expected)"
+                    .format(self.group.group_id, states, expected_states))
+                raise TransientRetryError(
+                    "Group states of {} did not match expected {})"
+                    .format(states, expected_states))
+
+            msg("Success: desired group state reached:\n{}"
+                .format(expected_states))
+            return self.rcs
+
+        def poll():
+            return self.get_stack_list().addCallback(check)
+
+        expected_states = pbag(expected_states)
+
+        return retry_and_timeout(
+            poll, timeout,
+            can_retry=terminal_errors_except(TransientRetryError),
+            next_interval=repeating_interval(period),
+            clock=reactor,
+            deferred_description=(
+                "Waiting for group {} to reach state {}".format(
+                    self.group.group_id, str(expected_states))))
+
+    @inlineCallbacks
+    def test_create(self):
+        """
+        For a launch_stack config, stacks are created, checked, updated, and
+        deleted through Heat.
+        """
+        p = ScalingPolicy(set_to=5, scaling_group=self.group)
+        scale_up = ScalingPolicy(set_to=7, scaling_group=self.group)
+        scale_down = ScalingPolicy(set_to=1, scaling_group=self.group)
+
+        yield self.group.start(self.rcs, self)
+
+        yield p.start(self.rcs, self)
+        yield p.execute(self.rcs)
+        yield self.wait_for_stack_list([u'UPDATE_COMPLETE'] * 5)
+
+        yield scale_up.start(self.rcs, self)
+        yield scale_up.execute(self.rcs)
+        yield self.wait_for_stack_list(
+            [u'UPDATE_COMPLETE'] * 5 + [u'CREATE_COMPLETE'] * 2)
```

Comment:
Only the final states of the stacks are captured here. In reality, the states from the first policy are:
```
1. [ ]
2. [CREATE_COMPLETE] * 5
3. [CHECK_COMPLETE] * 5
4. [UPDATE_COMPLETE] * 5
```
Going from 0 to N stacks is unusual because there's no `CHECK_` stacks to signal that it's the middle of a set of convergence cycles, so a check/update round is done at the end.

During the second policy execution, these are the states:
```
1. [UPDATE_COMPLETE] * 5
2. [CHECK_COMPLETE] * 5
3. [CHECK_COMPLETE] * 5 + [CREATE_COMPLETE] * 2
4. [UPDATE_COMPLETE] * 5 + [CREATE_COMPLETE] * 2
```
This is a more ordinary scenario (stacks already exist) so the stack checking is done at the beginning. Since there are no stacks in bad states in (3), Otter does stack update to clear the `CHECK_` status and convergence ends.

Finally, it is assured that scaling down to 1 stack will end up with one `UPDATE_COMPLETE` stack because stack checking is done at the beginning and all of the stacks will be in `CHECK_COMPLETE`. All but one are deleted and the remaining one is bumped to `UPDATE_COMPLETE`.